docs/src/apps/jsonrpc-api.md

```diff
@@ -395,12 +395,35 @@ The signatures will be ordered based on the Slot in which they were confirmed in
 
 ```bash
 // Request
-curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedSignaturesForAddress","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc": "2.0","id":1,"method":"getConfirmedSignaturesForAddressDepricated","params":["6H94zdiaYfRfPfKjYLjyr2VFBg6JHXygy84r3qhc3NsC", 0, 100]}' localhost:8899
 
 // Result
 {"jsonrpc":"2.0","result":{["35YGay1Lwjwgxe9zaH6APSHbt9gYQUCtBWTNL3aVwVGn9xTFw2fgds7qK5AL29mP63A9j3rh8KpN1TgSR62XCaby","4bJdGN8Tt2kLWZ3Fa1dpwPSEkXWWTSszPSf1rRVsCwNjxbbUdwTeiWtmi8soA26YmwnKD4aAxNp8ci1Gjpdv4gsr","4LQ14a7BYY27578Uj8LPCaVhSdJGLn9DJqnUJHpy95FMqdKf9acAhUhecPQNjNUy6VoNFUbvwYkPociFSf87cWbG"]},"id":1}
 ```
 
+### getConfirmedSignaturesForAddress
```

Comment:
@CriesofCarrots  - check out the latest `getConfirmedSignaturesForAddress`.  I think this'll be much nicer on clients, you paginate based on signature instead of slots at all (like who cares about slots) -- this'll be trivial to implement with the long-term transaction history proposal at https://github.com/solana-labs/solana/pull/10900.  Implementing the shorter history with the current rocksdb CFs will be inefficient but we can rework the CFs later if this becomes painful

Reply:
Also `limit` now becomes the actual limit :). (the default/max 1,000 limit directly maps to the Object list restrictions for cloud buckets)

Reply:
Yes, this does seem nicer.
Fyi, this approach will make the blockstore implementation slightly more complex, as we will first need to query the TransactionStatus cf for that signature to find out the slot to kick of the AddressSignatures iterator. But that's at most 2 reads, so I think nbd.